idaes/core/util/model_diagnostics.py

```diff
@@ -1229,6 +1244,10 @@ def report_structural_issues(self, stream=None):
         warnings, next_steps = self._collect_structural_warnings()
         cautions = self._collect_structural_cautions()
 
+        eval_error_warnings, eval_error_cautions = self._collect_potential_eval_errors()
```

Comment:
1. Is there an easy way this could be moved inside the `_collect_structural_warnings` and `_collect_structural_cautions` methods? It looks like you did this for efficiency so you only need to walk the model once. However, there is an `assert_no_structural_issues` method as well (that only checks for warnings) which could not include this as written (although the solution might be to just add this code there as well). I am also working on adding diagnostics to the convergence tester (parameter sweep) tool, and I am looking at using these methods there too (which would mean they need to become public).
2. Could you add an entry to `next_steps` as well pointing users to the associated `display` method.

Reply:
I think I have addressed both of these now. I realized I was being silly with the cautions. They were completely unnecessary/wrong.